### ISSUE TITLE
`Bloch` instance errors when trying render vectors

### ISSUE THREAD

**Issue by tomohiro-soejima** [qutip/qutip#1817]
### Bug Description

Call to `b.render()` errors after performing `b.add_vectors(vec)` where `b` is an instance of `qutip.Bloch()`.

### Code to Reproduce the Bug

```shell
import qutip
b = qutip.Bloch()
b.make_sphere()
b.add_vectors([0,0,1])
b.render()
```

### Code Output

```shell
/tmp/ipykernel_377526/3029128797.py in <module>
      2 b.make_sphere()
      3 b.add_vectors([0,0,1])
----> 4 b.render()

~/.conda/envs/myenv/lib/python3.8/site-packages/qutip/bloch.py in render(self)
    503         self.plot_annotations()
    504         # Trigger an update of the Bloch sphere if it is already shown:
--> 505         self.fig.canvas.draw()
    506 
    507     def plot_back(self):

~/.local/lib/python3.8/site-packages/matplotlib/backends/backend_agg.py in draw(self)
    434              (self.toolbar._wait_cursor_for_draw_cm() if self.toolbar
    435               else nullcontext()):
--> 436             self.figure.draw(self.renderer)
    437             # A GUI class may be need to update a window using this draw, so
    438             # don't forget to call the superclass.

~/.local/lib/python3.8/site-packages/matplotlib/artist.py in draw_wrapper(artist, renderer, *args, **kwargs)
     71     @wraps(draw)
     72     def draw_wrapper(artist, renderer, *args, **kwargs):
---> 73         result = draw(artist, renderer, *args, **kwargs)
     74         if renderer._rasterizing:
     75             renderer.stop_rasterizing()

~/.local/lib/python3.8/site-packages/matplotlib/artist.py in draw_wrapper(artist, renderer)
     48                 renderer.start_filter()
     49 
---> 50             return draw(artist, renderer)
     51         finally:
     52             if artist.get_agg_filter() is not None:

~/.local/lib/python3.8/site-packages/matplotlib/figure.py in draw(self, renderer)
   2808 
   2809             self.patch.draw(renderer)
-> 2810             mimage._draw_list_compositing_images(
   2811                 renderer, self, artists, self.suppressComposite)
   2812 

~/.local/lib/python3.8/site-packages/matplotlib/image.py in _draw_list_compositing_images(renderer, parent, artists, suppress_composite)
    130     if not_composite or not has_images:
    131         for a in artists:
--> 132             a.draw(renderer)
    133     else:
    134         # Composite any adjacent images together

~/.local/lib/python3.8/site-packages/matplotlib/artist.py in draw_wrapper(artist, renderer)
     48                 renderer.start_filter()
     49 
---> 50             return draw(artist, renderer)
     51         finally:
     52             if artist.get_agg_filter() is not None:

~/.local/lib/python3.8/site-packages/mpl_toolkits/mplot3d/axes3d.py in draw(self, renderer)
    449                                     for axis in self._get_axis_list()) + 1
    450                 collection_zorder = patch_zorder = zorder_offset
--> 451                 for artist in sorted(collections_and_patches,
    452                                      key=do_3d_projection,
    453                                      reverse=True):

~/.local/lib/python3.8/site-packages/mpl_toolkits/mplot3d/axes3d.py in do_3d_projection(artist)
    424                 """
    425                 try:
--> 426                     signature = inspect.signature(artist.do_3d_projection)
    427                     signature.bind()
    428                 # ValueError if `inspect.signature` cannot provide a signature

AttributeError: 'Arrow3D' object has no attribute 'do_3d_projection'
```

### Expected Behaviour

The code runs without an error.

### Your Environment

```shell
QuTiP Version:      4.6.3
Numpy Version:      1.22.2
Scipy Version:      1.8.0
Cython Version:     None
Matplotlib Version: 3.5.1
Python Version:     3.8.12
Number of CPUs:     16
BLAS Info:          OPENBLAS
OPENMP Installed:   False
INTEL MKL Ext:      False
Platform Info:      Linux (x86_64)
```

### Additional Context

This seems to be caused by [this](https://github.com/matplotlib/matplotlib/issues/21688) update in matplotlib. The solution seems to be to add the following method to `Arrow3D`.

```python
def do_3d_projection(self, renderer=None):
        xs3d, ys3d, zs3d = self._verts3d
        xs, ys, zs = proj3d.proj_transform(xs3d, ys3d, zs3d, self.axes.M)
        self.set_positions((xs[0],ys[0]),(xs[1],ys[1]))

        return np.min(zs)
```

**Comment by hodgestar** [comment]
@tomohiro-soejima Thank you for a very well organized bug report and for suggesting a fix. I've opened PR #1818 that applies your suggestion and adds tests. The PR is a continuation of #1690 because I didn't want to generate merge conflicts, so it can only be easily reviewed and merged once #1690 lands.

I've marked the fix for inclusion in 4.7 which should be released in the next couple of weeks.

**Comment by tomohiro-soejima** [comment]
Great, thanks for taking a look!
